benches/poseidon_benches.rs

```diff
@@ -0,0 +1,103 @@
+#[macro_use]
+extern crate criterion;
+extern crate hades252;
+
+use criterion::black_box;
+use criterion::Criterion;
+use rand::thread_rng;
+
+use bulletproofs::r1cs::{LinearCombination, Verifier};
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+
+use hades252::linear_combination;
+use hades252::scalar;
+
+
+mod scalar_benches {
+    use super::*;
+
+    /// This function allows us to bench the whole process of
+    /// digesting a vec of `Scalar`. We put together the process
+    /// of creating the scalar array + digesting the info to be able
+    /// to compare the results against the unoptimized version.
+    #[inline]
+    fn digest_one() -> () {
+        let s = Scalar::random(&mut thread_rng());
+        scalar::hash(&[s]).unwrap();
+    }
+
+    #[inline]
+    fn digest(n: usize) -> () {
+        let s: Vec<Scalar> = (0..n).map(|_| Scalar::random(&mut thread_rng())).collect();
+        scalar::hash(&s).unwrap();
+    }
+
+    pub fn hash_one_scalar(c: &mut Criterion) {
+        c.bench_function("Hashing single `Scalar`", |b| b.iter(|| digest_one()));
+    }
+
+    pub fn hash_two_scalars(c: &mut Criterion) {
+        c.bench_function("Hashing two `Scalar`", |b| b.iter(|| digest(black_box(2))));
+    }
+
+    pub fn hash_four_scalars(c: &mut Criterion) {
+        c.bench_function("Hashing four `Scalar`", |b| b.iter(|| digest(black_box(4))));
+    }
+}
+
+mod lc_benches {
+    use super::*;
+
+    /// This function allows us to bench the whole process of
+    /// digesting a vec of `LinearCombination`. We put together the process
+    /// of creating the linearcombination array + digesting the info to be able
+    /// to compare the results against the unoptimized version.
+    #[inline]
+    fn digest_one() -> () {
+        let s = LinearCombination::from(Scalar::random(&mut thread_rng()));
+
+        // The creation of the `Verifier` does not affect at the benchmark timings.
+        // It takes aproximately: `625.21 ns`.
```

Comment:
Nit: it doesn't make much sense add a value, since it's based on the hardware the test did run – and we don't have the target machine yet.

So it might be misleading: my suggestion is, remove it or replace with a percentage.
Something like "It takes around 0.1% of the total time" (or whatever is).
It's still not necessary true for all hardware, but it's less specific than the actual time, and give a better idea how much impact on tests (625.21 ns have a different weight if the total tests take 2 seconds or 40 seconds).

Reply:
Ohh I see. 
Yes, taking in mind that the verifier takes `ns` while the hash takes `ms`, it will be something around 0,000001%

So I'll add something like: `// It takes approximately a 0,000001% of the total amount of time that the whole hashing process represents.`

Ok for you @ZER0 ?

Reply:
I would say:
```
     // The `Verifier` instantiation takes roughly 0.000001% of the total benchmark time, so it's negligible
```

What do you think? But anything is good, as soon as it's documented! :)

Reply:
It's perfect, added. 

Once it passes the tests I'll merge and erase the branch!